Extension/lib/settings/settings-provider.js

```diff
@@ -0,0 +1,156 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Adguard Browser Extension.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var LS = require('../../lib/utils/local-storage').LS;
+var FilterLSUtils = require('../../lib/utils/filters-storage').FilterLSUtils;
+var ServiceClient = require('../../lib/utils/service-client').ServiceClient;
+var AntiBannerFiltersId = require('../../lib/utils/common').AntiBannerFiltersId;
+
+/**
+ * Application settings provider.
+ *
+ * @type {{loadSettingsManifest, loadFiltersSection, saveSettingsManifest, saveFiltersSection}}
+ */
+var SettingsProvider = (function () { // jshint ignore:line
+
+    var PROTOCOL_VERSION = "1.0";
+    var APP_ID = "adguard-browser-extension";
+
+    var SYNC_SETTINGS_TIMESTAMP_KEY = 'sync.settings.timestamp';
+    var SYNC_SETTINGS_FILTERS_TIMESTAMP_KEY = 'sync.settings.filters.timestamp';
+
+    var WHITELIST_DOMAINS_KEY = 'white-list-domains';
+    var DEFAULT_WHITELIST_FLAG_KEY = 'default-whitelist-mode';
+
+    var serviceClient = new ServiceClient();
+
+
+    var loadSettingsManifest = function () {
+        var manifest = {
+            "timestamp": LS.getItem(SYNC_SETTINGS_TIMESTAMP_KEY),
+            "protocol-version": PROTOCOL_VERSION,
+            "min-compatible-version": PROTOCOL_VERSION,
+            "app-id": APP_ID,
+            "sections": [
+                {
+                    "name": "filters.json",
+                    "timestamp": LS.getItem(SYNC_SETTINGS_FILTERS_TIMESTAMP_KEY)
+                }
+            ]
+        };
+
+        return manifest;
+    };
+
+    var loadFiltersSection = function (callback) {
+        var filtersState = FilterLSUtils.getFiltersStateInfo();
+        var enabledFilterIds = [];
+        for (var i = 0; i < filtersState.length; i++) {
+            var f = filtersState[i];
+            if (f && f.enabled) {
+                enabledFilterIds.push(i);
+            }
+        }
+
+        var whitelistDomains = [];
+        var json = LS.getItem(WHITELIST_DOMAINS_KEY);
+        if (json) {
+            whitelistDomains = JSON.parse(json);
+        }
+
+        var defaultWhiteListMode = LS.getItem(DEFAULT_WHITELIST_FLAG_KEY);
+
+        var onUserFilterRulesLoaded = function (result) {
+            var userFilterRules = [];
+
+            for (var i = 0; i < result.length; i++) {
+                var r = result[i];
+                userFilterRules.push(r.ruleText);
+            }
+
+            var section = {
+                "filters": {
+                    "enabled-filters": enabledFilterIds,
+                    "custom-filters": [
+                        // Custom filters are not supported yet
+                    ],
+                    "user-filter": {
+                        "rules": userFilterRules,
+                        "disabled-rules": ""
+                    },
+                    "whitelist": {
+                        "inverted": defaultWhiteListMode != 'true',
```

Comment:
`inverted` whitelist and simple whitelist are different.
You should have both here plus `whitelistMode` field.